### PR TITLE
MAINT: Replace MaskedArrayFutureWarning with FutureWarning.

### PR DESCRIPTION
**Pull request by charris**
The added specificity seems unnecessary, best to keep things simple.

### CODE REVIEW

**Comment by charris** [comment]
@jakirkham Do you have a strong reason to prefer `MaskedArrayFutureWarning`?

**Comment by jakirkham** [comment]
As there isn't really a way to opt-in to the new behavior for `__setitem__`, this provided users a way to silence warnings related to this change in `MaskedArray`s independently from silencing other `FutureWarning`s if they are confident that parts or all of their codebase would be unaffected by the change.

File: numpy/ma/core.py [hunk 1]
```diff
@@ -82,10 +82,7 @@
 MaskType = np.bool_
 nomask = MaskType(0)
 
-class MaskedArrayFutureWarning(FutureWarning):
-    pass
-
-
+class MaskedArray
```
Comment (jakirkham) [inline numpy/ma/core.py]: Probably meant to be removed.

Comment (charris) [inline numpy/ma/core.py]: My, how did that slip in there ?

**Comment by charris** [comment]
OK, well, let's leave it alone then.

**Pull request closed without merging**